### ISSUE TITLE
About the training strategy

### ISSUE THREAD

**Issue by SparkJiao** [AetherCortex/Llama-X#3]
Very nice project and appreciate your contribution!

I have seen the deepspeed config and I want to confirm the current training strategy. For LLaMA-13B, the training uses Zero-3 optimization, checkpointing, and CPU-offload, right? I'm curious if you have tried tensor parallel (used in original LLaMA training) or model parallel?

We would also love to contribute to the training implementation about model parallel for fast large scale training, aiming at models greater than 13B. Currently I'm investigating torch/fairscale pipeline parallel mechanism.

Best,
Fangkai

**Comment by AetherCortex** [comment]
Thank you for your interest and attention to our project. We have achieved model parallelism based on zero-3. Because the memory of V100 is relatively small, model parallelism is the key to training larger models(i.e., > 13B). Currently, there are some bugs when using this code on multi-node systems with 33B and 65B, and we welcome contributors to help solve this issue. If you can implement multi-node training with 33B and 65B based on this codebase, you can check in your code, and after our review & verification, we can merge it into the main code.

**Comment by haorannlp** [comment]
@AetherCortex  Hi, may I ask if batch size 64 is too larget for 8 V100 GPUs? I tried bs=64, gradient_update=1; bs=32, gradient_update=2; both have OOM error. All the other training settings follow this repo. Any suggestions?

**Comment by AetherCortex** [comment]
> @AetherCortex Hi, may I ask if batch size 64 is too larget for 8 V100 GPUs? I tried bs=64, gradient_update=1; bs=32, gradient_update=2; both have OOM error. All the other training settings follow this repo. Any suggestions?

The specific batchsize definitely depends on your environment, as long as everything is correct, this number should not be much different.

**Comment by tetelias** [comment]
> Thank you for your interest and attention to our project. 
> 
> For LLaMA-13B, the training uses Zero-3 optimization, checkpointing, and CPU-offload. Currently, there is a NCCL Error when using this code on multi-node systems with 33B and 65B. We also tried (used in original LLaMA training) model parallel, but there also exists a gradient explosion on multi-node sometimes. We are trying to solve these issues. 
> 
> And we welcome contributors to help solve this issue. If you can implement stable multi-node training with 33B and 65B based on this codebase or any other framework, you can check in your code, and after our verification, we can merge it into the main code.
Does V100 support bf16? It is obviously much more numerically stable than fp16 though I don't understand in what way does multi-node training would differ from single-node. Does it switch precision in optimizer to reduce amount of information passed?

**Comment by tetelias** [comment]
> Thank you for your interest and attention to our project. 
> 
> For LLaMA-13B, the training uses Zero-3 optimization, checkpointing, and CPU-offload. Currently, there is a NCCL Error when using this code on multi-node systems with 33B and 65B. We also tried (used in original LLaMA training) model parallel, but there also exists a gradient explosion on multi-node sometimes. We are trying to solve these issues. 
> 
> And we welcome contributors to help solve this issue. If you can implement stable multi-node training with 33B and 65B based on this codebase or any other framework, you can check in your code, and after our verification, we can merge it into the main code.

Does V100 support bf16? It is obviously much more numerically stable than fp16 though I don't quite understand in what way would multi-node training differ from single-node. Maybe it switches precision in optimizer to reduce amount of information passed?

**Comment by KelleyYin** [comment]
> Thank you for your interest and attention to our project. 
> 
> For LLaMA-13B, the training uses Zero-3 optimization, checkpointing, and CPU-offload. Currently, there is a NCCL Error when using this code on multi-node systems with 33B and 65B. We also tried (used in original LLaMA training) model parallel, but there also exists a gradient explosion on multi-node sometimes. We are trying to solve these issues. 
> 
> And we welcome contributors to help solve this issue. If you can implement stable multi-node training with 33B and 65B based on this codebase or any other framework, you can check in your code, and after our verification, we can merge it into the main code.

Have you solved this problem of training on multi-node systems?

**Comment by huiyangzhou** [comment]
> Thank you for your interest and attention to our project. 
> 
> For LLaMA-13B, the training uses Zero-3 optimization, checkpointing, and CPU-offload. Currently, there is a NCCL Error when using this code on multi-node systems with 33B and 65B. We also tried (used in original LLaMA training) model parallel, but there also exists a gradient explosion on multi-node sometimes. We are trying to solve these issues. 
> 
> And we welcome contributors to help solve this issue. If you can implement stable multi-node training with 33B and 65B based on this codebase or any other framework, you can check in your code, and after our verification, we can merge it into the main code.

do you finetune a dataset on llama-13B? i try many times , the result looks bad.  while i finetune a dataset on llama-7B , the result always looks  good?
